### ISSUE TITLE
Configure a video element for the "media" option and use the "peaks" option, the "ready" event not works well

### ISSUE THREAD

**Issue by shawchen08** [katspaugh/wavesurfer.js#3034]
## Bug description
Configure an video element for the "media" option and use the "peaks" option, the "ready" event not works well:
1. "ready" callback only executed at the first time, after refresh the page the callback not executed,  may be affected by the video cache
2. after removed the "peaks" option, "ready" callback works well

## Environment
 - Browser: Chrome

## Minimal code snippet
[Please see this demo](https://codesandbox.io/s/a-bug-of-the-ready-event-of-wavesurfer-js-6vqf45)

code:
```html
<script type="module">
  import WaveSurfer from "https://unpkg.com/wavesurfer.js@7/dist/wavesurfer.esm.js";

  fetch("./data.json")
    .then((res) => res.json())
    .then((res) => {
      const wavesurfer = WaveSurfer.create({
        container: document.body,
        waveColor: "rgb(200, 0, 200)",
        progressColor: "rgb(100, 0, 100)",

        peaks: [res.data],
        media: document.querySelector("video")
      });

      wavesurfer.on("interaction", () => {
        wavesurfer.play();
      });

      wavesurfer.on("ready", () => {
        // Change the text below the video
        document.getElementById("ready").innerText = "ok";
      });
    });
</script>
```

## Expected result
the "ready" event should works well

## Obtained result
the "ready" event only works well at the first time

**Comment by shawchen08** [comment]
The problem persists at v7.0.5, or gets worse, These codes are never executed:
![image](https://github.com/katspaugh/wavesurfer.js/assets/25865352/6c4841fe-14c3-4a7f-9ae4-c5f06ca76993)

After debugging, I found that wavesurfer fires the "ready" event before I listen to

I added a "ready" option to replace the "ready" even, it works

**Comment by katspaugh** [comment]
It should be fixed in 7.0.6 which I've just published.
